### PR TITLE
Fix symbol regex and format json file

### PR DESCRIPTION
**Pull request by salamaashoush**
Fix symbol regex to avoid matching ) or ] as part of the symbol

### CODE REVIEW

File: syntaxes/pact.tmLanguage.json [hunk 1]
```diff
@@ -1,350 +1,347 @@
...
+        {
+          "match": "([\\w%#+-_&\\$@<>=\\?\\*!\\|/]+)"
+        }
+      ]
+    },
+    "symbol": {
+      "patterns": [
+        {
+          "begin": "'",
+          "end": "(?=[\\s)|\\[\\],:\"'.])",
```
Comment (salamaashoush) [inline syntaxes/pact.tmLanguage.json:310]: this is the actual change the rest is formating

**Review by alber70g — APPROVED** [review]

**Review by Randynamic — APPROVED** [review]

**Comment by trendzetter** [comment]
This should have been done in 2 separate format and fix. Plus it should have been explained which error has been fixed. This is a murky approach to change

**Comment by trendzetter** [comment]
Mind you that the single quota syntax was actually already fixed here: https://github.com/kadena-community/pact-vscode/commit/1ea72b5775355808eff71d24d5dc7146123fe353
If it wasn't working for you you were probably running an older release (eg from the VSCode marketplace).

If i'm not mistaken all this change does is making it more difficult to read by adding unnecessary escape sequences? 

@salamaashoush

**Comment by salamaashoush** [comment]
> Mind you that the single quota syntax was actually already fixed here: [1ea72b5](https://github.com/kadena-community/pact-vscode/commit/1ea72b5775355808eff71d24d5dc7146123fe353) If it wasn't working for you you were probably running an older release (eg from the VSCode marketplace).
> 
> afaik all this change does is making it more difficult to read by adding unnecessary escape sequences?
> 
> @salamaashoush

It was not fixed since I was also running a local version, not the marketplace one, the escape characters are needed to escape [] since they are parsed as a character set in regex not an "]" or "[" and that's was the issue, the old regex was assuming that those are part of the symbol eg `['symbol]` the grammar was treating the last `]` as part of symbol unless if you add space before

**Comment by trendzetter** [comment]
the closing ] was already in the character class. It doesn't need escaping if it is the first character in a character class, as was the case.

**Comment by trendzetter** [comment]
"You can include an unescaped closing bracket by placing it right after the opening bracket, or right after the negating caret. []x] matches a closing bracket or an x." source: https://www.regular-expressions.info/charclass.html

**Comment by salamaashoush** [comment]
Thanks for the link, that's true regarding the behavior of the [], but the present and previous patterns have distinct behaviors. The current version groups the characters into a single set anyway this not the point, I'm seeking to comprehend whether the grammar is now malfunctioning for you, or if your sole concern is the code formatting and readability concern because of escape characters

**Comment by trendzetter** [comment]
The change is just not needed and the way it was applied -mixing formatting with attempts at fixing something- is bad practice. If there is a "distinct behavior" it should be possible to describe it.

**Pull request merged**